code/modules/mob/living/carbon/xenomorph/castes/hivemind/abilities_hivemind.dm

```diff
@@ -1 +1,177 @@
-// empty file for now
+// ***************************************
+// *********** Reposition Core
+// ***************************************
+/datum/action/xeno_action/activable/reposition_core
+	name = "Reposition Core"
+	action_icon_state = "emit_neurogas"
+	mechanics_text = "Reposition your core to a target location on weeds. The further away the location is the longer this will take up to a maximum of 60 seconds. 3 minute cooldown."
+	ability_name = "reposition core"
+	plasma_cost = 150
+	cooldown_timer = 180 SECONDS
+	keybind_flags = XACT_KEYBIND_USE_ABILITY
+	keybind_signal = COMSIG_XENOABILITY_REPOSITION_CORE
+
+/datum/action/xeno_action/activable/reposition_core/on_cooldown_finish()
+	owner.playsound_local(owner.loc, 'sound/voice/alien_drool1.ogg', 50, TRUE)
+	to_chat(owner, "<span class='xenonotice'>We regain the strength to reposition our neural core.</span>")
+	return ..()
+
+/datum/action/xeno_action/activable/reposition_core/use_ability(atom/A)
+	var/mob/living/carbon/xenomorph/hivemind/X = owner
+	var/obj/effect/alien/hivemindcore/core = X.core
+	var/turf/target_turf = get_turf(X)
+	var/distance = get_dist(target_turf, get_turf(core))
+
+	var/delay = max(HIVEMIND_REPOSITION_CORE_DELAY_MIN, HIVEMIND_REPOSITION_CORE_DELAY_MOD * distance) //Calculate the distance scaling delay before we complete the reposition
+
+
+	if(!check_build_location(target_turf, X)) //Check target turf for suitability
+		return fail_activate()
+	to_chat(owner, "<span class='xenodanger'>We begin the process of transfering our consciousness... We estimate this will require [delay * 0.1] seconds.</span>")
+
+	succeed_activate()
+	if(!do_after(X, delay, TRUE, null, BUSY_ICON_BUILD))
+		if(!QDELETED(src))
+			to_chat(X, "<span class='xenodanger'>We abort transferring our consciousness, expending our precious plasma for naught.</span>")
+			return fail_activate()
+
+	if(!check_build_location(target_turf, X)) //Check target turf for suitability after completion
```

Comment:
then should be replaced by can_use_ability() checks

Reply:
This is done.